src/machine/machine_stm32f407.go

```diff
@@ -154,7 +135,7 @@ func (uart UART) Configure(config UARTConfig) {
 	// Enable USART2 port.
 	uart.Bus.CR1.Set(stm32.USART_CR1_TE | stm32.USART_CR1_RE | stm32.USART_CR1_RXNEIE | stm32.USART_CR1_UE)
 
-	// Enable RX IRQ.
+	// Enable RX IRQ. TODO: pick the right IRQ_xxx for the bus and the uart
 	intr := interrupt.New(stm32.IRQ_USART2, func(interrupt.Interrupt) {
 		UART1.Receive(byte((UART1.Bus.DR.Get() & 0xFF)))
 	})
```

Comment:
This should probably live on the `UART1` definition, so it's more explicit about `USART2` interrupts being tied up there.